Refactoring and bug fixes

krekoten opened the pull request:
Hi! I've refactored code, switched active_support to 3.0.0 version. So it works with rails 3. If you like this changes, please pull them.
Thx.

krekoten committed:
Refactor code.

Move each model to separate file
Switched ActiveSupport to 3.0.0
Removed ActiveSupport monkeypatch
Moved URI patch to it's own file

krekoten committed:
Load correct file

krekoten committed:
Update site and token for autoloaded models

krekoten committed:
Fix updating auth credentials for autoloaded models.

krekoten committed:
Compute site late when we have all data.

caged commented:
Awesome work!  I'll merge this in next week.

caged commented:
I pulled down your changes and were testing them out, but there looks to be a problem with autoloading.  No matter what I attempt to run after the initial setting of your account and token, I'm receiving the error below:
         > Lighthouse::Base
        NameError: uninitialized constant Lighthouse::Base
            from ./lighthouse/project_membership.rb:2
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activesupport-3.0.0/lib/active_support/core_ext/module/introspection.rb:70:in `const_get'
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activesupport-3.0.0/lib/active_support/core_ext/module/introspection.rb:70:in`local_constants'
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activesupport-3.0.0/lib/active_support/core_ext/module/introspection.rb:70:in `each'
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activesupport-3.0.0/lib/active_support/core_ext/module/introspection.rb:70:in`local_constants'
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activesupport-3.0.0/lib/active_support/core_ext/module/introspection.rb:68:in `each'
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activesupport-3.0.0/lib/active_support/core_ext/module/introspection.rb:68:in`local_constants'
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activesupport-3.0.0/lib/active_support/core_ext/module/introspection.rb:86:in `local_constant_names'
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activesupport-3.0.0/lib/active_support/callbacks.rb:2
            from /Users/justin/.rvm/rubies/ruby-1.8.7-p302/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in`gem_original_require'
            from /Users/justin/.rvm/rubies/ruby-1.8.7-p302/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `require'
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activemodel-3.0.0/lib/active_model/validations/callbacks.rb:1
            from /Users/justin/.rvm/rubies/ruby-1.8.7-p302/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in`gem_original_require'
            from /Users/justin/.rvm/rubies/ruby-1.8.7-p302/lib/ruby/site_ruby/1.8/rubygems/custom_require.rb:31:in `require'
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activemodel-3.0.0/lib/active_model/validations.rb:7
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activeresource-3.0.0/lib/active_resource/validations.rb:63
            from /Users/justin/.rvm/gems/ruby-1.8.7-p302@rails3/gems/activeresource-3.0.0/lib/active_resource/base.rb:1408
            from ./lighthouse/base.rb:2

krekoten committed:
Require lighthouse/base.rb as workaround for ruby < 1.9.

ActiveSupport tries to emulate behavior of constants(false)
under ruby < 1.9, but it seems buggy.

krekoten commented:
This is active_support issue under ruby < 1.9. As quick workaround I've added `require 'lighthouse/base'`. I am working primarily under 1.9.2 so I didn't find this issue. Now it works both under 1.9 and 1.8.
